bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/transactions/TransactionResourceManager.java

```diff
@@ -186,6 +190,48 @@ private String getTransactionLogDirectory() {
         }
     }
 
+    /**
+     * This method gets the user specified config for the transaction auto commit timeout. Default is 120.
+     *
+     * @return int transaction auto commit timeout value
+     */
+    public static int getTransactionAutoCommitTimeout() {
+        VariableKey transactionAutoCommitTimeoutKey = new VariableKey(TRANSACTION_PACKAGE_ID,
+                TRANSACTION_AUTO_COMMIT_TIMEOUT_KEY, PredefinedTypes.TYPE_INT, false);
+        Object value = ConfigMap.get(transactionAutoCommitTimeoutKey);
+        if (value == null) {
+            return DEFAULT_TRX_AUTO_COMMIT_TIMEOUT;
+        }
+        return parseTimeoutValue(value, DEFAULT_TRX_AUTO_COMMIT_TIMEOUT);
+    }
+
+    /**
+     * This method gets the user specified config for cleaning up dead transactions. Default is 600.
+     *
+     * @return int transaction cleanup after value
+     */
+    public static int getTransactionCleanupTimeout() {
+        VariableKey transactionCleanupTimeoutKey = new VariableKey(TRANSACTION_PACKAGE_ID,
+                TRANSACTION_CLEANUP_TIMEOUT_KEY,
+                PredefinedTypes.TYPE_INT, false);
+        Object value = ConfigMap.get(transactionCleanupTimeoutKey);
+        if (value == null) {
+            return DEFAULT_TRX_CLEANUP_TIMEOUT;
+        }
+        return parseTimeoutValue(value, DEFAULT_TRX_CLEANUP_TIMEOUT);
+    }
+
+    private static int parseTimeoutValue(Object value, int defaultValue) {
+        if (!(value instanceof Number number)) {
+            return defaultValue;
+        }
+        int timeoutValue = number.intValue();
+        if (timeoutValue <= 0) {
+            return defaultValue;
+        }
+        return timeoutValue;
+    }
```

Comment:
```suggestion
private static int parseTimeoutValue(Object value, int defaultValue) {
    if (value instanceof Number number) {
        int timeoutValue = number.intValue();
        if (timeoutValue > 0) {
            return timeoutValue;
        }
    }
    return defaultValue;
}

```

Reply:
IMO, I think returning the parsed value at the bottom of the method makes it clear that the method’s primary purpose is to return a valid timeout value if possible, and only return the default value if the parsing or validation fails.

Reply:
The advantage of this refactoring is this reduces the number of return statements and keeps the code concise.